### ISSUE TITLE
Can not use authentication plugin

### ISSUE THREAD

**Issue by DmitrySokolov** [conan-io/conan#1129]
`pluginbase` module in `load_authentication_plugin()` is used incorrectly. Pip modules imported in a plugin file are lost.

MWE:

```python
# ----- example/plugins/plugin.py -----

import six


def get_object():
    return MyPlugin()


class MyPlugin(object):
    def __init__(self):
        self.var1 = 10

    def action(self):
        print("Action: old '{}'".format(self.var1))
        print("Enter new:")
        self.var1 = six.moves.input()
        print("Action: new '{}'".format(self.var1))


# ----- example/app.py -----

import os


# Bad plugin usage
def load_plugin(plugin_name):
    try:
        from pluginbase import PluginBase
        plugin_base = PluginBase(package="app.plugins")
        plugins_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "plugins")
        plugin_source = plugin_base.make_plugin_source(searchpath=[plugins_dir])
        return plugin_source.load_plugin(plugin_name).get_object()
    except:
        print("Error loading plugin '{}'".format(plugin_name))
        raise


# Good plugin usage
class App(object):
    def __init__(self, plugin_name):
        try:
            from pluginbase import PluginBase
            plugin_base = PluginBase(package="app.plugins")
            plugins_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "plugins")
            self.plugin_source = plugin_base.make_plugin_source(searchpath=[plugins_dir])
            self.plugin = self.plugin_source.load_plugin(plugin_name).get_object()
        except:
            print("Error loading plugin '{}'".format(plugin_name))
            raise


def main():
    a = App('plugin')
    a.plugin.action()
    print("-" * 20)

    p = load_plugin('plugin')
    p.action()


if __name__ == '__main__':
    main()
```

```text
Action: old '10'
Enter new:
21
Action: new '21'
--------------------
Action: old '10'
Enter new:
Traceback (most recent call last):
  File "C:/Dev/Projects/Test/PyPlugin/app.py", line 42, in <module>
    main()
  File "C:/Dev/Projects/Test/PyPlugin/app.py", line 38, in main
    p.action()
  File "C:\Dev\Projects\Test\PyPlugin\plugins/plugin.py", line 19, in action
    self.var1 = six.moves.input()
AttributeError: 'NoneType' object has no attribute 'moves'
```

**Comment by memsharded** [comment]
Confirmed. The module object is collected if a reference is not maintained to it, so it is necessary to hold it, in a App object, or maybe returning it:
```python
return plugin_source, plugin_source.load_plugin(plugin_name).get_object()
```

I might try to have a look to it later, maybe it could enter a minor release.

**Comment by memsharded** [comment]
Fixed and released in 0.21.1. Thanks very much for reporting and giving the solution!
